context/app/static/js/pages/CellsAPIDemo/CellExpressionHistogram/CellExpressionHistogram.jsx

```diff
@@ -0,0 +1,50 @@
+import React, { useState, useEffect } from 'react';
+import Typography from '@material-ui/core/Typography';
+
+import Histogram from 'js/shared-styles/charts/Histogram';
+import CellsService from '../CellsService';
+
+function CellExpressionHistogram({ uuid, geneName }) {
+  const [geneExpressionData, setGeneExpressionData] = useState([]);
+  const [diagnosticInfo, setDiagnosticInfo] = useState({});
+
+  useEffect(() => {
+    async function fetchCellExpression() {
+      const t0 = performance.now();
+
+      const response = await new CellsService().getCellExpressionInDataset({
+        uuid,
+        geneNames: [geneName],
+      });
+      const t1 = performance.now();
+      const timeWaiting = (t1 - t0) / 1000;
+      const numCells = response.length;
+      setDiagnosticInfo({ numCells, timeWaiting });
+      setGeneExpressionData(response.map((d) => d.values[geneName]));
+    }
+    fetchCellExpression();
+  }, [uuid, geneName]);
+
+  return geneExpressionData.length ? (
+    <>
+      <Typography>
+        {diagnosticInfo.timeWaiting.toFixed(2)} seconds to receive an API response for {diagnosticInfo.numCells} cells.
+      </Typography>
+      <Histogram
+        parentHeight={500}
+        parentWidth={500}
+        visxData={geneExpressionData}
+        margin={{
+          top: 50,
+          right: 50,
+          left: 50,
+          bottom: 50,
+        }}
+      />
+    </>
+  ) : (
+    <Typography>Please wait for histogram...</Typography>
```

Comment:
Maybe this is a place holder, but down the road, we probably want either a spinner, or a grey-box ghost?

Reply:
Yes, that will come down the road.